app/store/credit.ts

```diff
@@ -0,0 +1,69 @@
+import { supabase } from "../components/hippo/supabase";
+import { StoreKey } from "../constant";
+import { createPersistStore } from "../utils/store";
+
+export const useCreditStore = createPersistStore(
+  { credit: 5 },
+  (set, get) => ({
+    async fetchCredit(userId: string, isAnonymous?: boolean) {
+      console.log("userId", userId);
+      let { data: creditQuery, error } = await supabase
+        .from("credits")
+        .select("*")
+        .eq("id", userId);
+      if (creditQuery?.length === 0) {
+        this.upsertCredit(userId, isAnonymous ? 5 : 200);
+      } else {
+        console.log("creditQuery", creditQuery);
+        set(() => ({
+          credit: creditQuery ? creditQuery[0]?.credit : 5,
```

Comment:
does it mean if a user is out of credit, the user will always have a refresh of 5 credits anything he/she logins? @duchuy31

Reply:
Nếu user hết credit thì buộc phải login để có thể tiếp tục chat ạ